### PR TITLE
Deception CI Status

### PR DESCRIPTION
**Pull request by jaelynlitz**

### CODE REVIEW

**Review by pelesh — COMMENTED** [review]
My guess is this is not deploying CI on deception yet?

A nitpicking suggestion to have a newline at the end of each file.

**Review by cameronrutherford — APPROVED** [review]

File: buildsystem/spack/deception/sbatch.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# spack develop --no-clone --path=$(pwd) exago@develop &&
+# spack develop --clone --force FORCE --path=$(pwd)/hiop hiop@develop &&
+# cd $(pwd)/hiop &&
+# git submodule update --init --recursive &&
+# cd - &&
+./buildsystem/spack/configure-modules.sh 64
+
+EXIT_CODE=$?
+# Required to trigger trap handler
+exit $EXIT_CODE
```
[the remark this answers is not in the document]
Comment (cameronrutherford) [reply buildsystem/spack/deception/sbatch.sh:56]: End of file fixing is really easy to enforce through pre-commit. Not a priority right atm, but after hackathon would be easy to do and apply across all files.

**Comment by jaelynlitz** [comment]
This is now deploying correctly to Deception! Although, build is hanging here: https://gitlab.pnnl.gov/exasgd/resolve-mirror/-/jobs/131310#L331

**Comment by jaelynlitz** [comment]
All 4 checks are passing now! Ready to merge after another look from @ryandanehy @cameronrutherford or @pelesh

File: buildsystem/spack/deception/sbatch.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+#SBATCH -A exasgd
```
Comment (cameronrutherford) [inline buildsystem/spack/deception/sbatch.sh:2]: Don't think you need this file in this PR, unless you also want to add the spack module pipeline functionality to this PR as well

**Review by cameronrutherford — APPROVED** [review]
LGTM - need to add Ascent arch to commit status IMO cc @ryandanehy

**Comment by pelesh** [comment]
@jaelynlitz, is anything actually running on deception? I cannot see any tests there.

**Comment by jaelynlitz** [comment]
> @jaelynlitz, is anything actually running on deception? I cannot see any tests there.

Yes, I see a passing build and test in the pipeline https://gitlab.pnnl.gov/exasgd/resolve-mirror/-/jobs/131384

File: buildsystem/spack/deception/sbatch.sh (hunk 2 above)
Comment (jaelynlitz) [reply buildsystem/spack/deception/sbatch.sh:2]: Ah yes, I will remove for now

**Review by pelesh — APPROVED** [review]

**Comment by pelesh** [comment]
> LGTM - need to add Ascent arch to commit status IMO cc @ryandanehy

Is this within the scope of this PR or we should create a separate issue?

**Comment by jaelynlitz** [comment]
> > LGTM - need to add Ascent arch to commit status IMO cc @ryandanehy
> 
> Is this within the scope of this PR or we should create a separate issue?

I can make an issue for that to be fixed later to keep this PR to just Deception

**Comment by jaelynlitz** [comment]
> > > LGTM - need to add Ascent arch to commit status IMO cc @ryandanehy
> > 
> > 
> > Is this within the scope of this PR or we should create a separate issue?
> 
> I can make an issue for that to be fixed later to keep this PR to just Deception

https://github.com/ORNL/ReSolve/issues/24

**Review by ryandanehy — APPROVED** [review]

**Pull request merged**